src/v/pandaproxy/schema_registry/service.cc

```diff
@@ -202,18 +203,40 @@ ss::future<> service::configure() {
       _ctx.smp_sg, [has_ephemeral_credentials](service& s) {
           s._has_ephemeral_credentials = has_ephemeral_credentials;
       });
-    co_await _controller->get_security_frontend().local().create_acls(
-      {security::acl_binding{
-        security::resource_pattern{
-          security::resource_type::topic,
-          model::schema_registry_internal_tp.topic,
-          security::pattern_type::literal},
-        security::acl_entry{
-          principal,
-          security::acl_host::wildcard_host(),
-          security::acl_operation::all,
-          security::acl_permission::allow}}},
-      5s);
+
+    std::vector<cluster::errc> err_vec;
+    // This could be any non-success errc
+    err_vec.push_back(cluster::errc::timeout);
+    auto create_acls_backoff = initial_create_acls_backoff;
+
+    while (err_vec[0] != cluster::errc::success) {
```

Comment:
That is fair, however, I'm struggling to figure out what that additional criteria would be.
I agree that putting in an abort source is necessary but I believe that is not what you are referring to.

Reply:
Yeh, good point about it being arbitrary. Assuming this loop is in the background, I could see that it makes sense to let it keep running indefinitely. At the very least though, the timeout should not double indefinitely or it would become useless (e.g. sleep(5days)). The abort source can then handle the case of getting out of the loop during shutdown.

Reply:
For now, I'll stick with an arbitrary timeout of 5s (it should not take the cluster more that that to create ACLs).

I am making this decision because I have tested what happens when we (1) background this loop or (2) background SR startup. The result of (1) is that the principal may not have permissions set by the time the first SR request comes in and that would lead to failures. The result of (2) is that the SR may not be ready by the time SR requests come in.